streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -98,6 +98,10 @@ public StateDirectory(final StreamsConfig config, final Time time, final boolean
             throw new ProcessorStateException(
                 String.format("state directory [%s] doesn't exist and couldn't be created", stateDir.getPath()));
         }
+        if (stateDirName.startsWith("/tmp")) {
```

Comment:
Wondering if this should go into `StreamsConfig` ?

Reply:
Actually maybe we should wrap it in an `if hasPersistentStores` so that users won't get this warning if they don't actually have any persistent state